src/include/homestore/btree/detail/btree_mutate_impl.ipp

```diff
@@ -42,6 +42,11 @@ btree_status_t Btree< K, V >::do_put(const BtreeNodePtr& my_node, locktype_t cur
         return ret;
     }
 
+    auto unlock_lambda = [this](const BtreeNodePtr& node, locktype_t& cur_lock) {
+        unlock_node(node, cur_lock);
+        cur_lock = locktype_t::NONE;
+    };
```

Comment:
This is just a helper to reduce too much of repetitive code where it keeps track of unlock and cur_lock variable.